NPM install results results in conflict for nextjs v12.3.0

# Reproduction steps

- clone repository https://github.com/amaralc/ac
- delete yarn.lock
- run npm install
- verify error in terminal


npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: ac@0.0.0
npm ERR! Found: next@12.3.1
npm ERR! node_modules/next
npm ERR!   next@"12.3.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer next@"12.2.0" from @module-federation/nextjs-mf@5.10.0
npm ERR! node_modules/@module-federation/nextjs-mf
npm ERR!   @module-federation/nextjs-mf@"^5.9.10" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/ac/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ac/.npm/_logs/2022-10-20T03_45_43_275Z-debug-0.log

--- a/packages/nextjs-mf/package.json
+++ b/packages/nextjs-mf/package.json
@@ -22,7 +22,7 @@
   },
   "peerDependencies": {
     "webpack": "5.45.1",
-    "next": "12.2.0",
+    "next": "12.3.0",
     "react": "^17 || ^18",
     "react-dom": "^17 || ^18"
   }

--- a/packages/nextjs-mf/package.json
+++ b/packages/nextjs-mf/package.json
@@ -22,7 +22,7 @@
   },
   "peerDependencies": {
     "webpack": "5.45.1",
-    "next": "12.3.0",
+    "next": "^12",
     "react": "^17 || ^18",
     "react-dom": "^17 || ^18"
   }

